modules/reactor/src/meshgenerators/TriPinHexAssemblyGenerator.C

```diff
@@ -0,0 +1,507 @@
+  for (unsigned int i = 0; i < 3; i++)
+  {
+    if (_has_rings[i])
+    {
+      for (unsigned int j = 0; j < (_ring_radii[i].size() + (_ring_intervals[i][0] > 1)); j++)
+        block_ids_new[i].push_back(_ring_block_ids[i].empty() ? j + 1 : _ring_block_ids[i][j]);
+      block_ids_new[i].push_back(_background_block_ids.empty() ? max_ring_radii_size + 1
+                                                               : _background_block_ids.back());
+    }
+    else
+    {
+      block_ids_new[i].push_back(_background_block_ids.empty() ? (max_ring_radii_size + 1)
+                                                               : _background_block_ids.front());
+      block_ids_new[i].push_back(_background_block_ids.empty() ? (max_ring_radii_size + 2)
+                                                               : _background_block_ids.back());
+    }
+  }
+  std::vector<std::unique_ptr<ReplicatedMesh>> meshes;
+  for (unsigned int i = 0; i < 3; i++)
+  {
+    meshes.push_back(buildSinglePinSection(_side_length,
+                                           _ring_offset,
+                                           _ring_radii[i],
+                                           _ring_intervals[i],
+                                           _has_rings[i],
+                                           _preserve_volumes,
+                                           _num_sectors_per_side,
+                                           _background_intervals,
+                                           block_ids_new[i],
+                                           _node_id_background_meta));
+    if (!_pin_id_name.empty())
+      meshes[i]->add_elem_integer(_pin_id_name, true, _pin_id_values[i]);
+    if (i > 0)
+    {
+      MeshTools::Modification::rotate(*meshes[i], 120.0 * (Real)i, 0, 0);
+      meshes[0]->stitch_meshes(*std::move(meshes[i]),
+                               OUTER_SIDESET_ID,
+                               OUTER_SIDESET_ID,
+                               TOLERANCE,
+                               /*clear_stitched_boundary_ids=*/true);
+    }
+  }
+
+  if (_assembly_orientation == AssmOrient::pin_up)
+    MeshTools::Modification::rotate(*meshes[0], 90, 0, 0);
+  else
+    MeshTools::Modification::rotate(*meshes[0], 270, 0, 0);
+  /* Add subdomain names */
+  for (const auto & block_info_pair : block_info)
+    meshes[0]->subdomain_name(block_info_pair.first) = block_info_pair.second;
+  if (_external_boundary_id > 0)
+    MooseMesh::changeBoundaryId(*meshes[0], OUTER_SIDESET_ID, _external_boundary_id, false);
+  if (!_external_boundary_name.empty())
+  {
+    meshes[0]->get_boundary_info().sideset_name(
+        _external_boundary_id > 0 ? _external_boundary_id : (boundary_id_type)OUTER_SIDESET_ID) =
+        _external_boundary_name;
+    meshes[0]->get_boundary_info().nodeset_name(
+        _external_boundary_id > 0 ? _external_boundary_id : (boundary_id_type)OUTER_SIDESET_ID) =
+        _external_boundary_name;
+  }
```

Comment:
need to remove thee extra sidesets before returning (the default name, OUTER_SIDESET_ID)

Reply:
OUTER_SIDESET_ID boundary should always exist so that the mesh can be used as input of PatternedHexMG.

Reply:
ideally each object should be independent.
maybe we should use the mesh metadata to store the outer boundary id here, and retrieve it when building patterns?

What do you think?

Reply:
I agree with you that ideally the external boundary should be a meta data. Update this would involve changes in many `Reactor` classes though. We may do this in a separate PR later. Thank you for the suggestion.